client/my-sites/plans-features-main/index.tsx

```diff
@@ -245,6 +245,10 @@ const PlansFeaturesMain = ( {
 	const getPlanTypeDestination = usePlanTypeDestinationCallback();
 
 	const longerPlanTermDefaultExperiment = useLongerPlanTermDefaultExperiment();
+	const reflectStorageSelectionInPlanPrices =
+		( isEnabled( 'plans/term-savings-price-display' ) ||
+			longerPlanTermDefaultExperiment.isEligibleForTermSavings ) &&
+		isInSignup;
```

Comment:
Looking at this - is it correct? I think the logic should be -> Reflect it for all cases, except for when `longerPlanTermDefaultExperiment.isEligibleForTermSavings` is true. 

Also `isEnabled( 'plans/term-savings-price-display' )` doesn't feel relevant. I would consider removing.

Reply:
Hi @chriskmnds thanks for being so on top of this, but I was actually futzing with some really strange calypso local dev issues 😆 

This was meant to be a temporary commit while I cleaned things up on my machine.

Reply:
Actually on second thought these conditions are probably worth discussing since they're effectively how we determine eligibility for term savings in `useEligibilityForTermSavingsPriceDisplay` 

https://github.com/Automattic/wp-calypso/blob/90dd8f2a474db25c52ba248e34110693691bf915/client/my-sites/plans-features-main/hooks/use-eligibility-for-term-savings-price-display.ts#L59-L63

Reply:
Hmm needs a bit of mulling over indeed 😬

Reply:
> Looking at this - is it correct? I think the logic should be -> Reflect it for all cases, except for when longerPlanTermDefaultExperiment.isEligibleForTermSavings is true.

Hmmm...I keep thinking about this, and I sense that I'm missing something important. Feel free to correct me there's anything I'm not seeing.

In my head, it seems like if we were assigned to an experiment and visited the logged in plans page, the storage add-ons might be disabled without the `isInSignup` condition. 🤔 

I'm going to test some things out now.

Reply:
@chriskmnds I ended up deleting two comments that added quite a bit of confusion once I understood what you were conveying.

I've edited https://github.com/Automattic/wp-calypso/pull/96520#discussion_r1872023157 with a bit more explanation. Hopefully that clarifies things